### ISSUE TITLE
feat: implement true-value and false-value to ASwitch and ACheckbox

### ISSUE THREAD

**Issue by Sandros94** [jd-solanki/anu#135]
Following [Vue's documentation](https://vuejs.org/guide/essentials/forms.html#value-bindings):
> [Checkbox](https://vuejs.org/guide/essentials/forms.html#checkbox-1)
> 
> ``` vue
> <input
>   type="checkbox"
>   v-model="toggle"
>   true-value="yes"
>   false-value="no" />
> ```
> true-value and false-value are Vue-specific attributes that only work with v-model. Here the toggle property's value will be set to 'yes' when the box is checked, and set to 'no' when unchecked. You can also bind them to dynamic values using v-bind:
> 
> ``` vue
> <input
>   type="checkbox"
>   v-model="toggle"
>   :true-value="dynamicTrueValue"
>   :false-value="dynamicFalseValue" />
> ```
There is a native Vue feature to assign specific strings to a checkbox if it is true or false.

This is especially useful to reduce the necessary code needed to work with tools such as [Color-Mode](https://color-mode.nuxtjs.org/). Where you need to assign a specific string to `colorMode.preference` to select the desired theme.

We could have this:
``` vue
<ASwitch v-model="colorMode.preference" true-value="dark" false-value="light" />
```
Instead of this:
``` vue
<template>
  <div>
    <ASwitch v-model="themeToggle" />
  </div>
</template>

<script setup>
const colorMode = useColorMode()
const themeToggle = ref(false)

watch(themeToggle, (newValue) => {
	colorMode.preference = newValue ? 'dark' : 'light'
})
</script>
```
---
Not to mention that reading the actual value will get the switch to it's correct position. (excluding the 'system' preference)
I'm still quite new to javascript, so I'm not confident enough to open a PR without knowing exactly what files I need to change.
For a small background on how I got to this issue you can read [this discussion](https://github.com/jd-solanki/anu/discussions/134).

**Comment by Sandros94** [comment]
I've forgot to point out [this issue](https://github.com/jd-solanki/anu/discussions/134#discussioncomment-5091298) too:
In short when the v-model changes quickly right after the page load the ASwitch doesn't change it's visual state, but the v-model variable is in the correct state.

Ex, for:
`const themeToggle = ref(colorMode.value === 'dark');` at page load is false because for a short period of time `colorMode.value = 'system'` but a few milliseconds later it updates to `'dark'`. This makes the `themeToggle` go from false to true, but the actual visuals of the toggle don't change.

**Comment by Sandros94** [comment]
So I wanted to test stuff:
``` vue
<template>
    <div class="my-auto">
        <label>
            <input type="checkbox" :value="trueValue" v-model="isChecked" @change="onChange" />
            {{ isChecked }}
        </label>
    </div>
</template>

<script setup>
const props = defineProps({
    trueValue: {
        type: [String, Number, Boolean],
        default: true,
    },
    falseValue: {
        type: [String, Number, Boolean],
        default: false,
    },
})

const emit = defineEmits(['update:modelValue'])

const onChange = (event) => {
    emit('update:modelValue', event.target.checked ? props.trueValue : props.falseValue)
}

const isChecked = ref(false)
</script>
```
While defining `trueValue` and `falseValue` inside this component was the only thing need other than the `@change`, atm I have no clue on how tick/untick when the value changes from outside.

**Comment by jd-solanki** [comment]
Sorry, I didn't noticed checkbox is also there

**Comment by jd-solanki** [comment]
@Sandros94 We have `value` prop for a checkbox that will act as true value. Can you please confirm?

**Comment by Sandros94** [comment]
@jd-solanki Yes I've read the pr about it with the comments, and looking at the code (for that little I know about js/ts) it looks good.

This might sound dumb, because I really feel like that and I'm new to this. But, could you point me to a doc (or shortly explain it to me) on how can I test new/updated components like this pr, before they get to a full package update? Do I only copy the new component's code and put in my nuxt test-env or should I copy the whole repo and place it somewhere (like inside the node_module folder)?

**Comment by jd-solanki** [comment]
Hi @Sandros94 

First of all let's be clear on your feature requests:
- [x] true false value for switch via mentioned PR and check out the demo [here](https://github.com/jd-solanki/anu/blob/main/docs/components/demos/switch/DemoSwitchCustomModelValue.vue#L13-L14)
- [x] You can change checkbox value when checked via `value` prop in show in [this](https://anu-vue.netlify.app/guide/components/checkbox.html#array) demo
- [ ] Your [this](https://github.com/jd-solanki/anu/issues/135#issuecomment-1442505777) comment

Regarding, last point in above three, two are done and one still to resolve, right?

For your third point, You need to use [computed](https://vuejs.org/guide/essentials/computed.html) property instead of `ref`. When you use `const val = ref(colorMode.value === 'dark')` only the initial value will be set. This means on the first execution it will set `const val = ref(false)` and that's it, value won't change until you update it via `val.value = true`. Hence, in your case you need computed property that calculates based on another thing:
```diff
- const val = ref(colorMode.value === 'dark')
+ const val = computed(() => colorMode.value === 'dark')
```

Now, with this change `val` will auto update when `colorMode` changes.

---

Finally, You won't able to try mentioned PR with your existing anu installation.  I hope you are familiar with updating packages. To try latest commits before release, please perform the below changes explained in [this](https://github.com/jd-solanki/anu/blob/main/docs/guide/getting-started/edge-releases.md) guide.

---

I hope this will resolve your query.

**Comment by Sandros94** [comment]
**TL;DR**: I'm closing this issue because the last point that remained uncovered I'm mostly sure that is not related to the ASwitch.

Today I've got some free time and went back on this.
First off I wanted to make sure that I've understood correctly how the `computed` property worked. Only that, this time I actually understood what `get` and `set` functions are and how to use them. Then I made the switch to be able to be set back to the current theme configured by the system.
Here is the final code:
```vue
<template>
    <ClientOnly>
        <template #fallback>
            <ASwitch v-model="off" off-icon="i-svg-spinners:clock"/>
        </template>
        <ASwitch v-model="switchTheme" off-icon="i-ph:sun" on-icon="i-ph:moon"/>
    </ClientOnly>
</template>

<script setup>
const colorMode = useColorMode();
const off = ref(false)
const switchTheme = computed({
    get: () => colorMode.value === 'dark',
    set: (value) => {
        if (colorMode.preference === 'system') {
            // Set the opposite of the current system value
            colorMode.preference = value ? 'dark' : 'light';
        } else {
            // Set the preference back to 'system'
            colorMode.preference = 'system';
        }
    }
});
</script>
```

Now, about that `<ClientOnly>`:
I'm not sure why I didn't think of this earlier, but since I only need to know the system's current theme after the page has fully loaded by the browser, I can simply wait for it using the built-in component provided by Nuxt.
Then I added a dummy ASwitch just to animate it with a spinning clock until the actual one was ready.
